### PR TITLE
[DOCUMENTATION] Refresh James "design goal page"

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

File: src/site/xdoc/server/advantages.xml [hunk 1]
```diff
@@ -31,13 +31,12 @@
...
-          <p>100% pure Java Mail server designed to be a complete and portable enterprise
-            mail engine solution.</p>
+          <p>100% pure Java Mail server designed to be a complete enterprise mail engine solution.</p>
 
-          <p>Supports currently available IETF protocols, including SMTP, LMTP, POP3, IMAP4, ManageSIEVE. We also plan to support JMAP.</p>
+          <p>Supports currently available IETF protocols, including SMTP, LMTP, POP3, IMAP4. We also plan to support JMAP. ManageSIEVE support is experimental.</p>
 
           <p>Apache James Server is able to store user and message data either in any JPA-compatible database,
-            in Cassandra (allowing fast, reliable, even real-time replicated storage) and on a file-system (Maildir).</p>
+            in Cassandra (allowing fast, reliable, even real-time replicated storage) in combinaison with an objectstore.</p>
```
Comment (ottoka) [inline src/site/xdoc/server/advantages.xml:40]: Might as well mention the type of object store here, so people have a better idea what to expect. How about: "optionally in combination with an S3 compatible object store."

File: src/site/xdoc/server/objectives.xml [hunk 2]
```diff
@@ -28,10 +28,24 @@
...
+      </i>  The mail system is able to handle both mail
+      transport and storage in a single server application. Apache James
+      works alone without the need for any other server or solution.</p>
+    <p>
+      <i>
+        <b>Customisable solution</b>
+      </i>  Apache James behaviour can be easily extended by the means of
+      several plugin extensions, for instance <a href="/howTo/mail-processing.html">mailets</a>,
+      <a href="/howTo/custom-listeners.html">Mailbox listenrs</a>, <a href="/howTo/custom-listeners.html">SMTP hooks</a>,
+      <a href="/howTo/custom-smtp-commands.html">SMTP commands</a>, or <a href="/howTo/custom-webadmin-routes.html">WebAdmin routes</a>,
```
Comment (ottoka) [inline src/site/xdoc/server/objectives.xml:41]: Should be "Mailbox listeners"

File: src/site/xdoc/server/advantages.xml (hunk 1 above)
Comment (chibenwa) [reply src/site/xdoc/server/advantages.xml:40]: Yes for this one agree.

We should be careful IMO of not giving too much technical details inside very generic documentation page that might apply to other James packagings as well...

File: src/site/xdoc/server/advantages.xml [hunk 3]
```diff
@@ -31,13 +31,12 @@
...
```
Comment (chibenwa) [inline src/site/xdoc/server/advantages.xml:39]:
```suggestion
            in Cassandra (allowing fast, reliable, even real-time replicated storage) in combinaison with an objectstore (S3 compatible API).</p>
```

File: src/site/xdoc/server/objectives.xml [hunk 4]
```diff
@@ -28,10 +28,24 @@
...
+        <b>Complete solution</b>
+      </i>  The mail system is able to handle both mail
+      transport and storage in a single server application. Apache James
+      works alone without the need for any other server or solution.</p>
+    <p>
+      <i>
+        <b>Customisable solution</b>
+      </i>  Apache James behaviour can be easily extended by the means of
+      several plugin extensions, for instance <a href="/howTo/mail-processing.html">mailets</a>,
+      <a href="/howTo/custom-listeners.html">Mailbox listenrs</a>, <a href="/howTo/custom-listeners.html">SMTP hooks</a>,
```
Comment (chibenwa) [inline src/site/xdoc/server/objectives.xml:40]:
```suggestion
      <a href="/howTo/custom-listeners.html">Mailbox listeners</a>, <a href="/howTo/custom-listeners.html">SMTP hooks</a>,
```

**Review by Arsnael — APPROVED** [review]

**Pull request merged**